PORTAL/jobs/_job_compile_bench.py

```diff
@@ -368,9 +368,12 @@ def build_compile_script(
     maybe_branch = branch or ''
     ref = _utils.check_shell_str(req.pyperformance_ref)
 
-    cpython_repo = _utils.quote_shell_str(bfiles.repos.cpython)
-    pyperformance_repo = _utils.quote_shell_str(bfiles.repos.pyperformance)
-    pyston_benchmarks_repo = _utils.quote_shell_str(bfiles.repos.pyston_benchmarks)
+    cpython_repo = bfiles.repos.cpython
+    pyperformance_repo = bfiles.repos.pyperformance
+    pyston_benchmarks_repo = bfiles.repos.pyston_benchmarks
+    _utils.check_shell_str(cpython_repo)
+    _utils.check_shell_str(pyperformance_repo)
+    _utils.quote_shell_str(pyston_benchmarks_repo)
```

Comment:
Isn't this equivalent?

Reply:
If we shell-quote these paths *here* (on the portal machine) they won't go through tilde expansion on the worker machine.  It's basically impossible to do tilde expansion on one machine for another machine -- not only are there the `/home` vs. `/User` differences on Linux vs. Mac, but technically, the home directory isn't even guaranteed to be the same as the user name (`root` going to `/root` is the common example).

Reply:
Yeah, this makes sense now.